.github/workflows/reusable_release_automation.yaml

```diff
@@ -0,0 +1,115 @@
+name: Promote and Trigger Prerelease
+
+# This workflow encompasses two main functions:
+# 1. It checks for the latest successful prerelease, excluding any labeled with "artifacts-pending" or "pre-release-failure",
+#    and promotes it to a full release if suitable. This action should consequently trigger any workflows
+#    that respond to the 'released' event type, ultimately invoking 'reusable_on_release.yaml'.
+#
+# 2. It then triggers the creation of a new prerelease based on specified criteria and included files,
+#    fully automating the release cycle management.
+#
+# This workflow should be triggered merely from the default branch.
+# Manual prerelease creation is fully supported for exceptional cases.
+
+# Usage from the caller workflow:
+# jobs:
+#   release_management:
+#     uses: newrelic/coreint-automation/.github/workflows/reusable_release_automation.yaml@v3
+#     secrets:
+#       bot_token: ${{ secrets.bot_token }}
+#       slack_channel: ${{ secrets.slack_channel }}
+#       slack_token: ${{ secrets.slack_token }}
+
+on:
+  workflow_call:
+    inputs:
+      block_endpoint:
+        description: This endpoint should respond status 200 for the promotion to continue.
+        required: false
+        type: string
+        default: https://newrelic.github.io/coreint-automation/automatic_release_enable
+    secrets:
+      bot_token:
+        description: "github token"
+        required: true
+      slack_channel:
+        description: "slack channel for sending a message in case of failure"
+        required: true
+      slack_token:
+        description: "slack token for sending the message"
+        required: true
```

Comment:
We could avail ourselves of this (breaking) change to update secrets names to match the ones used in all the integrations (usually uppercase) and then use the action with `secrets: inherit`. WDYT?

Reply:
Good point! Updated in 4b2e6e0f22427753e337821f4ef0b190c1db4fde